text/0507-subnets.md

```diff
@@ -0,0 +1,402 @@
+
+      // targets must implement the IRouter interface
+      target: natGateway,
+    }),
+  ],
+});
+
+const privateSubnet = vpc.addSubnet('privateSubnet', {
+  cidrBlock: '10.2.0.0/20',
+  availabilityZone: 'us-west-2a',
+  routeTable,
+});
+```
+
+You can also create the same kind of routing pattern, but with a NAT instance
+instead:
+
+```ts
+const natInstance = publicSubnet.addNatInstance('natinst', {
+  instanceType: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.MICRO),
+  machineImage: ec2.MachineImage.latestAmazonLinux({
+    generation: ec2.AmazonLinuxGeneration.AMAZON_LINUX_2,
+  }),
+
+  // Other properties ommitted
+});
+
+const routeTable = vpc.addRouteTable('routeTable', {
+  routes: [
+    Route.to({
+      destination: '0.0.0.0/0',
+
+      // Instance also implements IRouter
+      target: natInstance,
+    }),
+  ],
+});
+
+const privateSubnet = vpc.addSubnet('privateSubnet', {
+  cidrBlock: '10.2.0.0/20',
+  availabilityZone: 'us-west-2a',
+  routeTable,
+});
+```
+
+For IPv6 traffic, you create this pattern by using an egress-only internet
+gateway:
+
+```ts
+const routeTable = vpc.addRouteTable('routeTable', {
+  routes: [
+    Route.local('10.0.0.0/16'),
+    Route.local('2001:db8:1234:1a00:/64'),
+
+    // The CIDR provided here must be for IPv6
+    Route.toEgressOnlyInternetGateway('::/0'),
+  ],
+});
+```
+
+To route traffic to a VPN Gateway, you can either explicitly add a route to the
+route table or you can enable route propagation:
+
+```ts
+const routeTable = vpc.addRouteTable('routeTable', {
+  routes: [
+    Route.local('10.0.0.0/16'),
+
+    // Static route to a VPN Gateway. Takes priority over propagated ones
+    // Causes VPN gateway to be enabled in the VPC
+    Route.toVpnGateway('172.31.0.0/24'),
+  ],
+
+  // To make VPN Gateway routes propagate to this route table
+  enableVpnGatewayRoutePropagation: true, // default: false
+});
+```
+
+If you have another VPC that you want to use in a peering connection:
+
+```ts
+const routeTable = vpc.addRouteTable('routeTable', {
+  routes: [
+    Route.local(),
+    Route.toPeerVpc({
+      vpc: anotherVpc, // or Vpc.fromLookup(...)
+      destination: '192.168.0.0/24', // The peer VPC CIDR
+    }),
+  ],
+});
+```
+
+Other targets include carrier gateways, transit gateways and network interfaces.
+The API to create routes to them follows the same pattern as above.
+
+### Using subnets with other components
+
+When you create a component that needs to be placed in a subnet, you can
+provide a subnet selection, which informs the `Vpc` construct which
+actual subnet to pick. On the other hand, if you already have an `ISubnet`
+instance where you want to place a component, you can provide it directly.
+For example, to create an `ApplicationLoadBalancer`:
+
+```ts
+const subnet1 = vpc.addSubnet(/*...*/);
+const subnet2 = vpc.addSubnet(/*...*/);
+
+const lb = new elbv2.ApplicationLoadBalancer(this, 'LB', {
+  vpc,
+  subnets: [subnet1, subnet2],
```

Comment:
`subnetSelection` already has these properties, we don't need to invent new ones.